### ISSUE TITLE
VimResized: Crash when redrawing after 2 lines of errors thrown

### ISSUE THREAD

**Issue by alemuller** [neovim/neovim#12432]
- `nvim --version`: NVIM v0.5.0-541-g8a1276005
- `vim -u DEFAULTS` (version: 8.2) behaves differently? Yes.
- Operating system/version: Arch Linux
- Terminal name/version:  kitty 0.17.4
- `$TERM`: xterm-kitty

### Steps to reproduce using `nvim -u NORC`

```
$ nvim -u NORC
# foo is undefined
:autocmd! VimResized * let &scroll = foo/4
# resize window
```

### Actual behaviour

The screen don't redraw, hitting CTRL-L crashes with the following message:
```
nvim: *nvim/message.c:2236: msg_scroll_flush: Assertion `pos_delta >= 0' failed.
```

**Comment by glacambre** [comment]
Can reproduce. What's interesting is that neovim doesn't crash 
immediately - I have to press a key to get it to crash.

**Comment by gelguy** [comment]
More details on the error: The cause seems to be throwing 2 lines of error in the `VimResized` autocmd. For the original example, the 2 lines are
```
E121: Undefined variable: foo
E15: Invalid expression foo/4
```
Other examples to reproduce the crash: `let foo/4`, `let &foo`.

Note that when an error is thrown in a `VimResized` autocmd the screen becomes blank. The crash seems to occur when the UI attempts to redraw the screen again.

Perhaps the title could be changed to be more relevant like VimResized: Crash when 2 lines of errors thrown.

**Comment by erratic-c-programmer** [comment]
It seems to go away when I remove the assert... Any reason why pos_delta must be non-negative?

**Comment by Adnios** [comment]
I solved this problem by "set display=lastline"

**Comment by zeertzjq** [comment]
The original steps to reproduce no longer produce the crash, but changing the autocommand to `:autocmd VimResized * cd /nonexistent` does.

**Comment by dundargoc** [comment]
Can't recreate the error. Anyone else that can?

**Comment by zeertzjq** [comment]
I can.

**Comment by dundargoc** [comment]
Huh. Weird. Tried "nvim --clean -u minimal.vim" with

":autocmd VimResized * cd /nonexistent" in minimal.vim but it didn't give any errors.

**Comment by zeertzjq** [comment]
Did you press a key after resize?

**Comment by dundargoc** [comment]
Aha, "resizing window" means the terminal window, not a vim window. OK, never mind, got it to work now.

**Comment by zeertzjq** [comment]
Fixed by #20760
